polaris-react/src/components/Frame/Frame.tsx

```diff
@@ -93,11 +100,52 @@ class FrameInner extends PureComponent<CombinedProps, State> {
     this.setOffset();
   }
 
-  componentDidUpdate(prevProps: FrameProps) {
+  componentDidUpdate(prevProps: FrameProps, prevState: State) {
     if (this.props.globalRibbon !== prevProps.globalRibbon) {
       this.setGlobalRibbonHeight();
     }
     this.setOffset();
+
+    if (this.state.polarisSummerEditions2023) {
+      if (
+        !prevState.showContextualSaveBar &&
+        this.state.showContextualSaveBar
+      ) {
+        let saveDisabled = false;
+        if (this.contextualSaveBar) {
+          if (this.contextualSaveBar.saveAction) {
+            if (this.contextualSaveBar.saveAction.disabled) {
+              saveDisabled = true;
+            }
+          }
+        }
```

Comment:
I believe this might give you the same result 

```suggestion
        const saveDisabled = this.contextualSaveBar?.saveAction?.disabled ?? false;
```